scripts/pages/settings/speedometer.js

```diff
@@ -674,21 +644,30 @@ class RangeColorProfiles {
 		RangeColorProfiles.clearProfiles();
 		RangeColorProfiles.markAsUnmodified();
 		RangeColorProfiles.keyvalues = SpeedometerSettingsAPI.GetColorProfiles();
-		for (const profileName of Object.keys(RangeColorProfiles.keyvalues)) {
-			RangeColorProfiles.objectList[profileName] = new RangeColorProfileObject(
+
+		for (const profile of RangeColorProfiles.keyvalues) {
+			// store in a map for fast indexing based on profile name
+			const profileName = profile[RangeColorProfileKeys.PROFILE_NAME];
+			RangeColorProfiles.profileObjectMap.set(
 				profileName,
-				RangeColorProfiles.keyvalues[profileName]
+				new RangeColorProfileObject(profileName, profile[RangeColorProfileKeys.PROFILE_RANGE_DATA])
 			);
 		}
 	}
 	static saveAllProfiles() {
-		const keyvalues = {};
-		for (const profileName of Object.keys(RangeColorProfiles.objectList)) {
-			keyvalues[profileName] = {};
-			const profileObject = RangeColorProfiles.objectList[profileName];
-			profileObject.saveToKV(keyvalues[profileName]);
+		const keyvalues = [];
+		for (const [profileName, profileObject] of this.profileObjectMap) {
+			const profileDataKV = [];
+			profileObject.saveToKV(profileDataKV);
+
+			const profile = {};
+			profile[RangeColorProfileKeys.PROFILE_NAME] = profileName;
+			profile[RangeColorProfileKeys.PROFILE_RANGE_DATA] = profileDataKV;
+			keyvalues.push(profile);
+
 			profileObject.markAsUnmodified();
```

Comment:
Another case for `.map`. Again if you make `saveToKV` return the array or object it's manipulating I think you can shorten this a great deal (I'm not sure this is exactly right):
```js
const keyvalues = this.profileObjectMap.map(([profileName, profileObject]) => {
  profileObject.markAsUnmodified();

  return {
    [RangeColorProfileKeys.PROFILE_NAME]: profileName,
    [RangeColorProfileKeys.PROFILE_RANGE_DATA]: profileObject.saveToKV([])
  };
});
```